generators/slack.py

```diff
@@ -301,18 +300,53 @@ def _calculate_top_teams_by_percentage(entities: list,
             for team, _ in sorted(entities_by_team.items(), key=lambda item: item[1], reverse=True)[:3]
         ]
         return top_3_teams_by_percentage
-
+    
     @staticmethod
-    def _generate_entities_list_with_level_and_link(blueprint: str,
-                                                    entities_by_level: Dict[str, List[Dict[str, str]]]) -> str:
-        text = ""
-        base_entity_url = f"{get_port_url(settings.port_region, 'app')}/{blueprint}Entity?identifier="
+    def _generate_entities_list_with_level_and_link(
+        blueprint: str,
+        entities_by_level: Dict[str, List[Dict[str, str]]]
+    ) -> list[dict[str, str]]:
+        """
+        Generates message block for Slack with entities grouped by level,
+        being aware of the 3001 message limit for Slack blocks.
+        """
+        block = []
         for level, entities in entities_by_level.items():
             if not entities:
                 continue
 
-            text += f"\n*{level}*\n\n"
-            for entity in entities:
-                text += f"• <{base_entity_url}{entity.get('identifier')}|{entity.get('name')}>" \
-                        f" - `[{len(entity.get('passed_rules'))}/{entity.get('number_of_rules')}] Passed` \n"
-        return text
+            batched_entities = utils.batch_items(
+                entities,
+                SlackMessageGenerator.SCORECARD_ENTITIES_BATCH_SIZE
+            )
```

Comment:
I feel like we need to batch them based on size of supported message size and not relying on number of entities. 
As we can have long entities names that will break the "20" entities that you have defined.